### PR TITLE
Allow setting default capacity for StringPool

### PR DESCRIPTION
**Pull request by carlhoerberg**
I think the previous default capacity of 8 entries is way too low for any read world use case, but maybe I'm wrong, can revert it if you want to.

### CODE REVIEW

**Comment by RX14** [comment]
allocating two half-page allocations doesn't seem that bad

File: spec/std/string_pool_spec.cr [hunk 1]
```diff
@@ -65,4 +65,12 @@ describe StringPool do
     end
     pool.size.should eq(10_000)
   end
+
+  it "can be created with larger initial capacity" do
```
Comment (yxhuvud) [inline spec/std/string_pool_spec.cr]: It is smaller with the other change, though.

**Comment by asterite** [comment]
So, String Pool is also used when parsing JSON to reuse object keys. I doubt there will be 256 different keys. Can we keep it small but make it configurable?

(I don't know how this change affects the performance of parsing JSON, though)

File: src/string_pool.cr [hunk 2]
```diff
@@ -33,8 +33,18 @@ class StringPool
...
+  # of the internal buffers in case of growth. If you have an estimate
+  # of the maximum number of elements the pool will hold it should
+  # be initialized with that capacity for improved performance.
+  #
+  # ```
+  # pool = StringPool.new(256)
+  # pool.size # => 0
+  # ```
+  def initialize(initial_capacity : Int = 8)
+    @capacity = initial_capacity
```
Comment (asterite) [inline src/string_pool.cr]: The type restriction is too wide and it won't work if you pass anything other than Int32. Let's make that Int32 instead.

File: spec/std/string_pool_spec.cr (hunk 1 above)
Comment (carlhoerberg) [reply spec/std/string_pool_spec.cr]: hehe, yes, but reverted back to 8 entries now, so now it's true again ;)

File: spec/std/string_pool_spec.cr [hunk 3]
```diff
@@ -65,4 +65,12 @@ describe StringPool do
...
+    pool = StringPool.new(capacity: 32)
```
Comment (asterite) [inline spec/std/string_pool_spec.cr]:
```suggestion
    pool = StringPool.new(initial_capacity: 32)
```

**Pull request merged**